src/core/server/saved_objects/migrations/core/build_active_mappings.ts

```diff
@@ -47,11 +49,36 @@ import {
  * @param typeDefinitions - the type definitions to build mapping from.
  */
 export function buildActiveMappings(
-  typeDefinitions: SavedObjectsTypeMappingDefinitions | SavedObjectsMappingProperties
+  typeDefinitions: SavedObjectsTypeMappingDefinitions | SavedObjectsMappingProperties,
+  opensearchDashboardsRawConfig?: Config
 ): IndexMapping {
   const mapping = defaultMapping();
 
-  const mergedProperties = validateAndMerge(mapping.properties, typeDefinitions);
+  let mergedProperties = validateAndMerge(mapping.properties, typeDefinitions);
+  // if permission control for saved objects is enabled, the permissions field should be added to the mapping
+  if (opensearchDashboardsRawConfig?.get('savedObjects.permission.enabled')) {
+    const principals: SavedObjectsFieldMapping = {
+      properties: {
+        users: {
+          type: 'keyword',
+        },
+        groups: {
+          type: 'keyword',
+        },
+      },
+    };
+    mergedProperties = validateAndMerge(mapping.properties, {
+      permissions: {
+        properties: {
+          read: principals,
+          write: principals,
+          management: principals,
```

Comment:
Discussed this with @zengyan-amazon, the `management` permission is essentially the same as `write` permission which allows the principals to update/delete the workspace object itself. In that sense, a user with `write` and `library_write` should be sufficient to make the user a workspace admin.

In the current implementation of workspace creation and workspace permission validation, we actually no longer rely on `management`. So we don't need to have it in the index mapping.

Reply:
make sense

Reply:
I've removed the management permission type yet, please help to take a look, thanks!